### PR TITLE
refactor(medusa): Migrate middlewares to ts

### PR DESCRIPTION
**Pull request by adrien2p**
**Goal**
- Migrate all middlewares to TS

**Improvements**
- Create a module augmentation over Express to use the merging interface system and add the following properties as part of the express `Request` type
    - `user`
    - `scope`

### CODE REVIEW

File: packages/medusa/src/api/middlewares/error-handler.ts [hunk 1]
```diff
@@ -51,7 +53,6 @@ export default () => {
         break
       default:
         errObj.code = "unknown_error"
-        errObj.message = "An unknown error occurred."
```
Comment (zakariaelas) [inline packages/medusa/src/api/middlewares/error-handler.ts:54]: question: is this replaced somewhere?

File: packages/medusa/src/api/middlewares/normalized-query.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { NextFunction, Request, Response } from "express"
+
```
Comment (zakariaelas) [inline packages/medusa/src/api/middlewares/normalized-query.ts:2]: suggestion (non-blocking, if-minor): Not really related to the PR @adrien2p so feel free to ignore, but I think it'd be cool to have a short comment here describing what this middleware does, wdyt?

Comment (zakariaelas) [reply packages/medusa/src/api/middlewares/normalized-query.ts:2]: maybe something like this: 
```typescript
/**
 * Middleware for parsing an array like query param to an array
 * e.g: /admin/orders/?fields[]=id,status,cart_id becomes { fields: ["id", "status", "cart_id"] }
 */
```

Let's ask @olivermrbl to validate the intent 😄

**Review by zakariaelas — APPROVED** [review]
Great work @adrien2p 🚀 ! LGTM 👍

File: packages/medusa/src/api/middlewares/error-handler.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/api/middlewares/error-handler.ts:54]: it just not overriding line 23 in fact 😂

**Review by olivermrbl — APPROVED** [review]
LGTM!

File: packages/medusa/src/api/middlewares/normalized-query.ts (hunk 2 above)
Comment (olivermrbl) [reply packages/medusa/src/api/middlewares/normalized-query.ts:2]: Correct me if I am wrong @adrien2p, but #1593 would still make use of `normalized-query`, so I believe adding a comment is reasonable no?

Comment (adrien2p) [reply packages/medusa/src/api/middlewares/normalized-query.ts:2]: 👍 ill do that it is fine for me

**Pull request merged**